sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -220,10 +236,10 @@ public Mono<SpringService> createResourceAsync() {
         if (isInCreateMode()) {
             createOrUpdate = manager().serviceClient().getServices()
                 .createOrUpdateAsync(resourceGroupName(), name(), innerModel());
-        } else if (needUpdate) {
-            needUpdate = false;
+        } else if (patchToUpdate != null) {
             createOrUpdate = manager().serviceClient().getServices().updateAsync(
-                resourceGroupName(), name(), innerModel());
+                resourceGroupName(), name(), patchToUpdate);
+            patchToUpdate = null;
```

Comment:
usually safer to init to `new ServiceResourceInner()`?

Reply:
Changed.